### PR TITLE
fixed 2 bugs reported on primary workspace

### PR DESCRIPTION
**Pull request by heejaechang**
1. filter out projects that is not supported in remote host when calculating checksum (open xaml designer)
2. wrong validation on contract (rename project)

both of them is crashing issue.

unit test added

...

this is a bug fix for this PR - https://github.com/dotnet/roslyn/pull/15212

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @srivatsn @KevinH-MS @tmat can you take a look?

**Comment by heejaechang** [comment]
@srivatsn can I get approval for RC3

File: src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal static class RemoteSupportedLanguages
+    {
+        public static bool Support(this string language)
```
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs]: sure

**Comment by heejaechang** [comment]
retest windows_eta_open_prtest please

File: src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal static class RemoteSupportedLanguages
```
Comment (KevinH-MS) [inline src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs]: consider: "Remote" in type name might be redundant in this namespace (not sure if it conflicts with another "SupportedLanguages" without though).

**Comment by KevinH-MS** [comment]
Seems reasonable to me 👍

File: src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs (hunk 2 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs]: I dont know what is better name. but without Remote, kind of conflict with other general ones. I will leave it like this for now.

**Comment by heejaechang** [comment]
@tannergooding I think the failure is real one?

**Comment by heejaechang** [comment]
windows_eta_open_prtest

**Comment by heejaechang** [comment]
retest windows_eta_open_prtest please

**Pull request merged**